### ISSUE TITLE
Random freezing

### ISSUE THREAD

**Issue by Anakael** [swaywm/sway#4199]
Hello! I have trouble with random freezing on sway. It independent on load or other reasons so it's hard to reproduce.

My sway version - 1.0
There is my log : https://pastebin.com/29jeTuWx
There is my config: https://pastebin.com/xhEfFHV4
Video driver: nouveau
Os: Arch Linux

Hope for help.

**Comment by RedSoxFan** [comment]
The debug log makes it looks like sway might be crashing, but not returning you to a tty. Are you using systemd or did you setup a different init system?

**Comment by Anakael** [comment]
I use systemd and run sway with sddm. Running with "sway" from TTY gives the same result.

**Comment by RedSoxFan** [comment]
Since you are using systemd, sway *should* return you to a gettty/dm on a segfault. I guess there is no harm in checking `coredumpctl gdb sway` anyway. 

Since you seem to be able to reproduce, is the following line (ignoring timestamp and pid) always the last line that sway itself logs?
```
2019-06-01 00:25:40 - [sway/tree/root.c:270] Recording workspace for process 11250
```

**Comment by Anakael** [comment]
There is a sway's dump https://pastebin.com/Txn4FCD3.
I'll check about "Recording workspace for process 11250".

**Comment by RedSoxFan** [comment]
> ```
> Timestamp: Sun 2019-05-19 22:49:44 UTC (1 weeks 6 days ago)
> ```

Unfortunately, that dump is probably not related to this issue

**Comment by Anakael** [comment]
There is one more log: https://pastebin.com/vy09zvYF

**Comment by Emantor** [comment]
This could be #4178, this log stops at applying transaction as well.

**Comment by Anakael** [comment]
I have checked log one more time and didn't found anythings related to transactions.

**Comment by blueStruct** [comment]
Hi, I have the same problem. It happens always ~1-10 mins after starting sway. The screen freezes and the sound runs in a 2-3 sec loop. Also I can't reach any ttys.

sway version: 1.1.1-3
logs: https://pastebin.com/trCMMhHU --- https://pastebin.com/xzKBfnZz
config: https://pastebin.com/J1Z3b0d5
video driver: mesa 19.1.3-1
os: Arch Linux, kernel 5.2.3
systemd version: 242.84-1

hope this helps. If you need more information or anything else, I would gladly help.

**Comment by thisch** [comment]
I have the same problem. (I'll add logs and more info after my vacation).

**Comment by blueStruct** [comment]
Sorry, please ignore my post, after some hours it also happened to me with Gnome on Xorg. In my case it is apparently not related to sway.

**Comment by travankor** [comment]
>Since you are using systemd, sway should return you to a gettty/dm on a segfault.

What magic does systemd do that other inits do not do here? Can other inits  be extended to do this with elogind?

Whenever sway freezes for me, the entire system becomes unresponsive and needs a forced shutdown (haven't looked into sysrq but not sure if it works on wayland/sway).

**Comment by BrendanBall** [comment]
It's also happening to me on Arch, very randomly when I'm just busy doing normal work.

**Comment by ascent12** [comment]
Take a look at your kernel logs. It may be a GPU reset, which is basically a way to say your GPU crashed.
Sometimes it can recover from this, but your session is still well and truly broken when this happens.

If it does turn out that this is the problem, there isn't really anything we can do to fix it though. It would be a driver or hardware issue.

**Comment by Anakael** [comment]
-> driver if hardware issue
In that case it must happens on another WM. But on awesome WM, for example - everything is fine.

**Comment by emersion** [comment]
>In that case it must happens on another WM. But on awesome WM, for example - everything is fine.

No. wlroots doesn't use the driver in the same way as xorg-xserver. wlroots may trigger a bug xorg-xserver doesn't trigger.

**Comment by surister** [comment]
Happening to me as well, 3-4 times in 2h while doing nothing speciall.
Everything freezes and I have to phisically hard reboot.
It's a fresh Arch install, everything as you may imagine, up to date.
Did not get any of this with i3-gaps

**Comment by Anakael** [comment]
Don't happening after swtich video card to raredon

**Comment by surister** [comment]
> Happening to me as well, 3-4 times in 2h while doing nothing speciall.
> Everything freezes and I have to phisically hard reboot.
> It's a fresh Arch install, everything as you may imagine, up to date.
> Did not get any of this with i3-gaps

Switched back again to i3-gaps, 0 issue, I guess I will try sway in a year.

**Comment by mizzunet** [comment]
Same here. I will grab logs on the next happening.

**Comment by ModProg** [comment]
Im having a similar issue, where sway hangs for about a minute and than continues working as if nothing happened.

I can also not switch to TTY2 with Alt+Ctrl+F2 while this is going on.

I have intel integrated graphics.

I have noticed some correlation with heavy computation loads like starting a Java compilation, but often times it happens just during a Teams meeting.

Using Arch with the latest version.

**Comment by johannes-titz** [comment]
Same here. But I can switch to another tty and kill sway. Also running Arch.

**Comment by TrudeEH** [comment]
I have the same issue. Sometimes sway freezes, and I can't do anything, except for moving the mouse. Also on Arch, with intel integrated graphics.
